Tests/L0/AzureRmWebAppDeployment/_suite.ts

```diff
@@ -0,0 +1,327 @@
+        
+        setResponseFile('armGoodWithParamFileUser.json');
+
+        var tr = new trm.TaskRunner('AzureRmWebAppDeployment');
+        tr.setInput('ConnectedServiceName', 'AzureRMSpn');
+        tr.setInput('WebAppName', 'mytestapp');
+        tr.setInput('Package', 'webAppPkg.zip');
+        tr.setInput('UseWebDeploy', 'true');
+        tr.setInput('SetParametersFile', 'invalidparameterFile.xml');
+
+        tr.run()
+            .then(() => {
+
+                assert(tr.invokedToolCount == 1, 'should have invoked tool once');
+                assert(tr.stderr.length > 0, 'should have written to stderr');
+                var expectedErr = 'Error: Set parameters file not found: invalidparameterFile.xml'; 
+                assert(tr.stdErrContained(expectedErr), 'should have said: ' + expectedErr);
+                assert(tr.failed, 'task should have succeeded');
+                done();
+
+            })
+            .fail((err) => {
+                done(err);
+            });
+    });
+
+    it('run AzureRmWebAppDeployment Task with missing SPN input', (done) => {
+        
+        setResponseFile('armGood.json');
+
+        var tr = new trm.TaskRunner('AzureRmWebAppDeployment');
+        tr.setInput('WebAppName', 'mytestapp');
+        tr.setInput('Package', 'webAppPkg.zip');
+        tr.setInput('UseWebDeploy', 'true');
+        
+        tr.run()
+            .then(() => {
+
+                assert(tr.invokedToolCount == 0, 'should not have invoked any tool');
+                assert(tr.stderr.length > 0, 'should have written to stderr');
+                var expectedErr = 'Input required: ConnectedServiceName'; 
+                assert(tr.stdErrContained(expectedErr), 'should have said: ' + expectedErr); 
+                assert(tr.failed, 'task should have failed');
+                done();
+
+            })
+            .fail((err) => {
+                done(err);
+            });
+    });
+
+    it('run AzureRmWebAppDeployment Task with missing WebAppName', (done) => {
+        
+        setResponseFile('armGood.json');
+
+        var tr = new trm.TaskRunner('AzureRmWebAppDeployment');
+        tr.setInput('ConnectedServiceName', 'AzureRMSpn');
+        tr.setInput('Package', 'webAppPkg.zip');
+        tr.setInput('UseWebDeploy', 'true');
+        
+        tr.run()
+            .then(() => {
+
+                assert(tr.invokedToolCount == 0, 'should not have invoked any tool');
+                assert(tr.stderr.length > 0, 'should have written to stderr');
+                var expectedErr = 'Input required: WebAppName'; 
+                assert(tr.stdErrContained(expectedErr), 'should have said: ' + expectedErr); 
+                assert(tr.failed, 'task should have failed');
+                done();
+
+            })
+            .fail((err) => {
+                done(err);
+            });
+    });
+
+    it('run AzureRmWebAppDeployment Task with missing Package Name Input', (done) => {
```

Comment:
'it' should describe what will happen after running this test. It should not be what test is doing. e.g. Fail for missing package name input. 
Ref - http://samwize.com/2014/02/08/a-guide-to-mochas-describe-it-and-setup-hooks/
https://github.com/Microsoft/vsts-tasks/blob/master/Tests/L0/SSH/_suite.ts#L96